tests/unit/test_solvers.py

```diff
@@ -0,0 +1,376 @@
+"""Test all solvers in :module:`coreax.solvers`."""
```

Comment:
The easiest check would be to compare the runtime of the integration tests - I will update with the results

Reply:
@tc85324 do you have results from this?

Reply:
This is a single run with no additional optimisations on a GPU enabled device. The New code is generally slightly faster, but this could simply be due to timing variance. Proper performance benchmarking will still be required to assess performance regressions in the future #10, #273, #510

OLD        | NEW        | Name
-----------| -----------| -----
21.38s     | ***20.29s*** | test_herding_basic.py
***14.29s*** | 15.91s     | test_pounce_map_reduce_ssm.py
15.27s     | ***14.47s*** | test_david_map_reduce_weighted.py
15.06s     | ***14.28s*** | test_herding_basic_weighted.py
13.51s     | ***13.25s*** | test_herding_refine.py
9.76s      | ***9.63s***  | test_pounce_map_reduce.py
8.01s      | ***7.68s***  | test_pounce.py
***5.13s***  | 5.96s      | test_herding_stein_weighted_ssm.py
5.34s      | ***4.65s***  | test_herding_stein_weighted.py
***2.48s***  | 3.26s      | test_herding_with_approximate_gram_matrix.py

Reply:
Thanks for this, Happy to consider this resolved and progress the listed tickets in the future.